woodwork/table_accessor.py

```diff
@@ -1124,6 +1124,19 @@ def infer_temporal_frequencies(self, temporal_columns=None):
             self._dataframe, temporal_columns=temporal_columns
         )
 
+    @_check_table_schema
+    def validate(self, return_indices=False):
+        if return_indices:
+            indices = [
+                self.ww[column].ww.validate(return_indices=True)
+                for column in self.columns
+            ]
+            return pd.concat(indices, axis=1)
+
+        else:
+            for column in self.columns:
+                self.ww[column].ww.validate(return_indices=False)
```

Comment:
this will only catch the first invalid column--maybe we want to offer an option that just returns the message or a True/False value that can be used like `is_valid`

Reply:
I don't think it's necessarily crucial for this initial pass. I see this more like a nice to have that probably needs more design. Maybe we can circle back if it becomes necessary?